### PR TITLE
DOP-1975: Add breadcrumb schema and simplify component

### PR DESCRIPTION
**Pull request by sophstad**
### Stories/Links:

DOP-1975

### Staging Links:

[Compass Staging](https://docs-mongodbcom-staging.corp.mongodb.com/master/compass/sophstad/DOP-1975/)

### Notes:

- Attach page titles (both rich & plaintext) in `gatsby-node.js` to minimize logic performed by the breadcrumb components
- Remove dependency on legacy `mongodb-docs.css` in breadcrumb render
- Will ask Croud team to perform acceptance testing before merging

### CODE REVIEW

File: __mocks__/gatsby.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ module.exports = {
   ...gatsby,
   graphql: jest.fn(),
   StaticQuery: jest.fn(),
-  withPrefix: jest.fn().mockImplementation((str) => str),
+  withPrefix: jest.fn().mockImplementation((path) => (path.startsWith(`/`) ? path : `/${path}`)),
```
Comment (sophstad) [inline __mocks__/gatsby.js:9]: Updated this mock to make it more similar to the actual Gatsby implementation [here](https://github.com/gatsbyjs/gatsby/blob/54d4721462b9303fed723fdcb15ac5d72e103778/packages/gatsby-link/src/index.js#L23); without this change, links generated in tests behaved a bit unexpectedly.

File: src/components/BreadcrumbSchema.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+const BreadcrumbSchema = ({ breadcrumb, siteTitle }) => {
+  const { siteUrl } = useSiteMetadata();
+  return (
+    <Helmet>
+      {Array.isArray(breadcrumb) && (
+        <script type="application/ld+json">
+          {JSON.stringify({
+            '@context': 'https://schema.org',
+            '@type': 'BreadcrumbList',
+            itemListElement: getBreadcrumbList([{ path: '/', plaintext: siteTitle }, ...breadcrumb], siteUrl),
```
Comment (sophstad) [inline src/components/BreadcrumbSchema.js:25]: Our breadcrumb data structure doesn't include the root page, so add it here to ensure it's captured in the schema.

**Review by rayangler — APPROVED** [review]
LGTM, aside from the lint warnings

File: src/components/BreadcrumbSchema.js (hunk 2 above)
Comment (sophstad) [reply src/components/BreadcrumbSchema.js:25]: No, I don't believe the current page should be included based on DevHub's breadcrumbs, which have already been Croud approved (you can see [this page](https://developer.mongodb.com/quickstart/nodejs-change-streams-triggers/) only includes breadcrumbs for home, learn, and quickstart). Will wait for confirmation from Croud on this, though!

**Review by casthewiz — APPROVED** [review]

**Review by rayangler — APPROVED** [review]

**Pull request merged**